### PR TITLE
[homekit] add BasicFan accessory

### PR DESCRIPTION
**Pull request by ccutrer**
This is Fan v1. It's a subset of FanV2, except that Home allows you to customize the icon to show a ceiling fan.

### CODE REVIEW

File: bundles/org.openhab.io.homekit/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <dependency>
       <groupId>io.github.hap-java</groupId>
       <artifactId>hap</artifactId>
-      <version>2.0.1</version>
+      <version>2.0.2-SNAPSHOT</version>
```
Comment (jlaur) [inline bundles/org.openhab.io.homekit/pom.xml:25]:
```suggestion
      <version>2.0.2</version>
```

probably? Seems released:
https://github.com/hap-java/HAP-Java/releases/tag/hap-2.0.2

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitBasicFanImpl.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import io.github.hapjava.accessories.BasicFanAccessory;
+import io.github.hapjava.characteristics.HomekitCharacteristicChangeCallback;
+import io.github.hapjava.services.impl.BasicFanService;
+
+/**
+ * Implements Fan using an Item that provides an On/Off state
+ *
+ * @author Eugen Freiter - Initial contribution
+ */
+class HomekitBasicFanImpl extends AbstractHomekitAccessoryImpl implements BasicFanAccessory {
```
Comment (jlaur) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitBasicFanImpl.java:33]: `@NonNullByDefault`?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks!

Build failed with `Error:  Failed to execute goal on project org.openhab.io.homekit: Could not resolve dependencies for project org.openhab.addons.bundles:org.openhab.io.homekit:jar:3.4.0-SNAPSHOT: Could not find artifact io.github.hap-java:hap:jar:2.0.2-SNAPSHOT in openhab-snapshot`

**Comment by ccutrer** [comment]
Yeah, sorry, I forgot I had built a snapshot locally. It actually needs 2.0.3. I'll update this PR when I get a chance.

**Comment by ccutrer** [comment]
This should be ready for review now, but should not be merged until https://github.com/openhab/openhab-addons/pull/13484 because the HAP-Java update will cause brokenness without that PR going first.

**Review by yfre — APPROVED** [review]
pretty straight forward. 
LGTM 
thank you @ccutrer

**Comment by lolodomo** [comment]
I suppose you have to release now that #13484 is merged? (the change of hap version is already done in the merged PR).

**Comment by ccutrer** [comment]
@lolodomo : no, nothing needs to be released.  The BasicFan support was already in HAP 2.0.4 (that openhab-addons started depending on in #13484). I also don't need to rebase or merge, since the "conflicting" diff is the same, so will just resolve itself.

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**